hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/http/HttpServer2.java

```diff
@@ -669,6 +674,16 @@ private void initializeWebServer(String name, String hostName,
     addDefaultApps(contexts, appDir, conf);
     webServer.setHandler(handlers);
 
+    // Jetty StatisticsHandler should be the first handler.
+    // The handler returns 503 if there is no next handler and the response is
+    // not committed. In Apache Hadoop, there are some servlets that do not
+    // commit (i.e. close) the response. Therefore the handler fails.
```

Comment:
If I understand correctly, this paragraph explains why we need to put the StatisticsHandler as the first handler, right? Is it possible that we can have a unit test to reproduce the scenario where the response is not committed ?

Reply:
I think it is possible. I'll create a unit test to reproduce the scenario.

Reply:
- I found why the handler should be inserted and updated the explanation: https://www.eclipse.org/lists/jetty-users/msg06273.html
- Created a test case for the scenario: https://github.com/apache/hadoop/pull/2145/commits/34bb0a6855cec3faca4a5b7eec323b5ca273743d

Thank you @Jing9  for your review.